packages/design-system-react/src/components/text/Text.types.ts

```diff
@@ -0,0 +1,173 @@
+   * @default TextColor.TextDefault
+   */
+  color?: TextColor;
+};
+export enum TextVariant {
+  // Display Sizes
+  DisplayLg = 'display-lg',
+  DisplayMd = 'display-md',
+
+  // Heading Sizes
+  HeadingLg = 'heading-lg',
+  HeadingMd = 'heading-md',
+  HeadingSm = 'heading-sm',
+
+  // Font Sizes
+  BodyLg = 'body-lg',
+  BodyMd = 'body-md',
+  BodySm = 'body-sm',
+  BodyXs = 'body-xs',
+}
+
+export enum TextColor {
+  /** For default neutral text. */
+  TextDefault = 'text-default',
+  /** For softer contrast neutral text */
+  TextAlternative = 'text-alternative',
+  /** For the softest contrast neutral text (not accessible) */
+  TextMuted = 'text-muted',
+  /** For elements used on top of overlay/alternative. Used for text, icon or border */
+  OverlayInverse = 'text-overlay-inverse',
+  /** For interactive, active, and selected semantics. Used for text, background, icon or border */
+  PrimaryDefault = 'text-primary-default',
+  /** For elements used on top of primary/default. Used for text, icon or border */
+  PrimaryInverse = 'text-primary-inverse',
+  /** For the critical alert semantic elements. Used for text, background, icon or border */
+  ErrorDefault = 'text-error-default',
+  /** For the stronger contrast error semantic elements. */
+  ErrorAlternative = 'text-error-alternative',
+  /** For elements used on top of error/default. Used for text, icon or border */
+  ErrorInverse = 'text-error-inverse',
+  /** For the positive semantic elements. Used for text, background, icon or border */
+  SuccessDefault = 'text-success-default',
+  /** For elements used on top of success/default. Used for text, icon or border */
+  SuccessInverse = 'text-success-inverse',
+  /** For the caution alert semantic elements. Used for text, background, icon or border */
+  WarningDefault = 'text-warning-default',
+  /** For elements used on top of warning/default. Used for text, icon or border */
+  WarningInverse = 'text-warning-inverse',
+  /** For informational read-only elements. Used for text, background, icon or border */
+  InfoDefault = 'text-info-default',
+  /** For elements used on top of info/default. Used for text, icon or border */
+  InfoInverse = 'text-info-inverse',
+  /** Inherit the color of the parent element */
+  Inherit = 'text-inherit',
+  /** Make the text color transparent */
+  Transparent = 'text-transparent',
+}
+
+export enum TextAlign {
+  Left = 'text-left',
+  Center = 'text-center',
+  Right = 'text-right',
+  Justify = 'text-justify',
+}
+
+export type ValidTag =
+  | 'dd'
+  | 'div'
+  | 'dt'
+  | 'em'
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6'
+  | 'li'
+  | 'p'
+  | 'span'
+  | 'strong'
+  | 'ul'
+  | 'label'
+  | 'input'
+  | 'header'
+  | 'a'
+  | 'button';
+
+export enum FontWeight {
+  /**
+   * Weight - 700
+   */
+  Bold = 'font-bold',
+  /**
+   * Weight - 500
+   */
+  Medium = 'font-medium',
+  /**
+   * Weight - 400
+   */
+  Normal = 'font-normal',
```

Comment:
Added comments to match Figma. See [slack conversation](https://consensys.slack.com/archives/C02VD8QG0LT/p1732140025025819)